queues.lisp

```diff
@@ -0,0 +1,67 @@
+;;;; Erlangen queue arsenal, these queues are multi-producer, single-consumer
+;;;; FIFO queues.
+
```

Comment:
@ruldolfochrist Hey there, I see you are hunting for concurrent queues (at least that’s what I am imagining).

Reply:
That looks really nice. Didn't know about _Vyukov queues_ before. Thanks for the tip! 😄

Reply:
AFAIU, if you need a bounded queue you will most likely end up with a ring buffer (they all look pretty similar, but if you want to go high-performance on multi-core architectures you end up optimizing the memory layout/contention, see [A lock-free, cache-efficient shared ring buffer for multi-core architectures](http://www.cse.cuhk.edu.hk/~pclee/www/pubs/ancs09poster.pdf)), for unbounded queues the Vyukov queue was the neatest I could find (can be nicely expressed in CL too).

Reply:
Btw, IIRC both queues below are designs from [Dimitry Vyukov](http://www.1024cores.net/) (awesome guy), I really should have attributed that properly.